src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurAccessTokenV2ServiceImpl.java

```diff
@@ -0,0 +1,166 @@
+package edu.cornell.kfs.concur.batch.service.impl;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status.Family;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.glassfish.jersey.client.ClientConfig;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider;
+
+import edu.cornell.kfs.concur.ConcurConstants;
+import edu.cornell.kfs.concur.ConcurConstants.ConcurOAuth2;
+import edu.cornell.kfs.concur.ConcurParameterConstants;
+import edu.cornell.kfs.concur.batch.service.ConcurAccessTokenV2Service;
+import edu.cornell.kfs.concur.batch.service.ConcurBatchUtilityService;
+import edu.cornell.kfs.concur.businessobjects.ConcurOauth2PersistedValues;
+import edu.cornell.kfs.concur.rest.jsonObjects.ConcurOauth2TokenResponseDTO;
+import edu.cornell.kfs.sys.service.WebServiceCredentialService;
+import edu.cornell.kfs.sys.util.CUJsonUtils;
+import edu.cornell.kfs.sys.util.CURestClientUtils;
+
+public class ConcurAccessTokenV2ServiceImpl implements ConcurAccessTokenV2Service {
+    private static final Logger LOG = LogManager.getLogger();
+    
+    protected WebServiceCredentialService webServiceCredentialService;
+    protected ConcurBatchUtilityService concurBatchUtilityService;
+
+    @Override
+    public String retrieveNewAccessBearerToken() {
+        ConcurOauth2PersistedValues credentialValues = getConcurOauth2PersistedValuesFromWebServiceCredentials();
+        ConcurOauth2TokenResponseDTO tokenResponse = getAccessTokenFromConcurEndPoint(credentialValues);
+        updateRefreshTokenIfRequired(credentialValues, tokenResponse);
+        return tokenResponse.getAccess_token();
+    }
+    
+    private ConcurOauth2PersistedValues getConcurOauth2PersistedValuesFromWebServiceCredentials() {
```

Comment:
You mentioned that "OAuth2" was the convention to use. Should the other methods and classes in this PR be updated likewise? Or would you prefer to delay that cleanup to a future user story?